### PR TITLE
[SYCL] Fix acos edge case

### PR DESCRIPTION
**Pull request by jzc**

### CODE REVIEW

**Review by KseniyaTikhomirova — APPROVED** [review]
SYCl E2E LGTM

File: sycl/test-e2e/Regression/acos.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  queue q;
+  const double pi = std::atan2(+0., -0.);
+  T data[][2] = {{{-2, -INFINITY}, {pi / 2, INFINITY}},
+                 {{-0., INFINITY}, {pi / 2, -INFINITY}}};
+  int N = std::size(data);
+  auto *p = malloc_shared<T>(N, q);
+
+  q.single_task([=] {
+     for (int i = 0; i < N; ++i) {
+       p[i] = std::acos(data[i][0]);
```
Comment (AlexeySachkov) [inline sycl/test-e2e/Regression/acos.cpp:21]: Note: we do not guarantee that `std::acos` works in kernels. But we should be supporting `::cacos` according to our [docs](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/C-CXX-StandardLibrary.rst).

Comment (jzc) [reply sycl/test-e2e/Regression/acos.cpp:23]: Interesting, it seems we are still testing the std::complex math functions in our testing though: https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/DeviceLib/std_complex_math_fp64_test.cpp

**Comment by AlexeySachkov** [comment]
Windows pre-commit failed due to seemingly infrastructure issue and we had a clean run of it before merge commit [here](https://github.com/intel/llvm/actions/runs/9550010455/job/26323842420), so I will proceed with merging this PR
